### PR TITLE
Fixes issue #105 : Users can view nfts without connecting wallet 

### PR DESCRIPTION
**Pull request by AmanSingh494**
Fixes issue #105 
### Problem
1. Users need to connect wallet to view nfts in exploreNft page.
2. Fetching nfts through pinata gateway from frontend is not secure.

### Solution
1. Used Infura provider instead of wallet to get wallet to fetch data from smart contracts.
2. Moved nft fetching logic from frontend to backend by making an api

### Changes
1. loadNfts functions runs when explorePage mounts without needing to connect wallet.
2. User need to connect wallet when buying an nft.
3. Added route in the backend to handle fetchNfts request.
4. Interaction with blockchain and IPFS from the backend.
5. Compressed images using Pako library and send to the frontend.

### Testing
I have tested the api and its working with a harcoded metadata cid and image cid for fetching data from IPFS as I don't have access to the admin's IPFS. 

### For Admin
1. Please review my changes and let me know of any error in the code or its working. I would try to solve it.
2. I have compressed the images before sending to frontend but more optimisations need to be done to reduce the load time.

### CODE REVIEW

**Comment by AmanSingh494** [comment]
Also you need to add the Infura url in the environment variables in the backend
`INFURA_URL='https://sepolia.infura.io/v3/<your api key>'`

File: frontend/src/utils/ethereum.js [hunk 1]
```diff
@@ -164,49 +163,30 @@ export const createNFT = async (signer, name, description, price, file) => {
...
-        //dbg
-        // console.log('Fetched metadata:', metadata)
-        return metadata
-    } catch (error) {
-        console.error('Error fetching metadata:', error)
-        return null
     }
+    const items = JSON.parse(result)
+    const data = items.data
+    data.forEach(async (item) => {
```
Comment (4darsh-Dev) [inline frontend/src/utils/ethereum.js:182]: the async function should have await expression
## Recommended
```javascript

async function fetchDataAsync(): Promise<string> {
  return await fs.readFile("data.txt", "utf-8")
}

performAction(async () => { await writeToFile(data) });

// Allow empty functions.
async function no_op() {}


```

**Review by 4darsh-Dev — APPROVED** [review]
fine, 👍🏻

**Comment by 4darsh-Dev** [comment]
Gentle Reminder 🚨🚨 
 GSSoC-Ext 24 is going to complete in 2 Days. '
 Complete your assigned Issues and PR reviews before time to get Points on Leaderboard. '
 It was a great experience working with you all, Don't forget to ⭐ star the Repo (only 12 with 62 forks 💔) . 
 Thanks💗 for your valuable contributions!

**Pull request merged**